### ISSUE TITLE
Cannot push to large repo

### ISSUE THREAD

**Issue by Jookia** [gitbucket/gitbucket#2502]
## Issue
**Impacted version**: git master with my pending PRs

**Deployment mode**: Standalone app with nginx proxy frontend

**Problem description**: I cloned the Linux kernel repo to GitBucket then pushed a few commits to it. This fails mid-way through, hanging after 'Updating references'

```
jookia_dev@novena-choice-citizen:/data/dev/linux% git push LS WIP_v5.8-LS
Enter passphrase for key '/home/jookia_dev/.ssh/id_ed25519': 
Enumerating objects: 24, done.
Counting objects: 100% (24/24), done.
Delta compression using up to 4 threads
Compressing objects: 100% (17/17), done.
Writing objects: 100% (17/17), 3.96 KiB | 11.00 KiB/s, done.
Total 17 (delta 13), reused 0 (delta 0)
remote: Resolving deltas: 100% (13/13)
remote: Updating references: 100% (1/1)
Connection to git.lumina-sensum.com closed by remote host.
fatal: the remote end hung up unexpectedly
fatal: the remote end hung up unexpectedly
git push LS WIP_v5.8-LS  0.85s user 4.23s system 0% cpu 10:39.74 total
```

**Comment by McFoggy** [comment]
did you adapted the nginx config, especially `client_max_body_size`

see https://github.com/gitbucket/gitbucket/wiki/Reverse-proxy-with-Nginx for some examples/samples of configuration

**Comment by xogium** [comment]
Yes, we did that. In fact, we cranked it up to 16 gb. We even increasted the read and write proxy timeout to 1000 instead of 100, which resulted in it hanging for 10 minutes then giving up.

**Comment by jphustman** [comment]
I think this is on the client side.  The server I am pushing from has only 2G of ram.  I've tried all sorts of stuff in the Nginx configuration, the one thing that finally worked was 

`git repack -a -d -f --window=0`

before pushing...  courtesy of  https://stackoverflow.com/a/57751121/1704342

**Comment by xogium** [comment]
Erm... Are you sure this is the right command ?

Cloned linux here, .git is 3.4 gb.
After running this command, .git is now 31 gb.
